experimental/unitsconversion/src/info_provider.rs

```diff
@@ -0,0 +1,160 @@
+    pub unit_quantity: VarZeroVec<'data, UnitQuantityULE>,
+
+    /// Contains the conversion information, such as the conversion rate and the base unit.
+    /// For example, the conversion information for the unit `foot` is `1 foot = 0.3048 meter`.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub convert_units: VarZeroVec<'data, ConvertUnitsULE>,
+}
+
+#[zerovec::make_ule(UnitsInfoIndexULE)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(databake::Bake),
+    databake(path = icu_unitsconversion::units_provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Copy, Debug, Clone, Default, PartialEq, PartialOrd, Eq, Ord)]
+pub struct UnitsInfoIndex {
+    /// Contains the index of the quantity in the `unit_quantity` vector.
+    /// If the unit does not have a quantity, this field is `None`.
+    pub quantity: Option<u16>,
+
+    /// Contains the index of the convert unit in the `convert_units` vector.
+    /// If the unit does not have a convert unit, this field is `None`.
+    pub convert_unit: Option<u16>,
+}
+
+#[zerovec::make_ule(QuantitySimplicityULE)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_unitsconversion::units_provider),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[derive(Copy, Clone, Debug, PartialOrd, Ord, PartialEq, Eq, Default)]
+#[repr(u8)]
+pub enum QuantitySimplicity {
+    #[default]
+    Simple = 0,
+    Complex = 1,
+}
+
+#[zerovec::make_varule(UnitQuantityULE)]
+#[derive(Clone, Debug, PartialOrd, Ord, PartialEq, Eq, Default)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(databake::Bake),
+    databake(path = icu_unitsconversion::units_provider),
+)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize),
+    zerovec::derive(Serialize)
+)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Deserialize),
+    zerovec::derive(Deserialize)
+)]
+#[zerovec::derive(Debug)]
+pub struct UnitQuantity<'data> {
+    /// Contains the quantity name.
+    // TODO(#4173): Consider using an enum for the quantity name.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub quantity: Cow<'data, str>,
+
+    /// Represents the simplicity of the quantity.
+    pub constant_exactness: QuantitySimplicity,
+}
+
+#[zerovec::make_varule(ConvertUnitsULE)]
+#[derive(Clone, Debug, PartialOrd, Ord, PartialEq, Eq, Default)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(databake::Bake),
+    databake(path = icu_unitsconversion::units_provider),
+)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize),
+    zerovec::derive(Serialize)
+)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Deserialize),
+    zerovec::derive(Deserialize)
+)]
+#[zerovec::derive(Debug)]
+pub struct ConvertUnits<'data> {
+    /// Contains the base unit which the unit is converted to.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub base_unit: Cow<'data, str>,
```

Comment:
could you use the index of the unit to avoid having a string here?

Reply:
+1; we should aim to include a measurement unit identifier string table that maps from the string identifier to a numeric identifier, because the measurement unit identifiers are large and storing variable-length items is slow.

Reply:
In this case, it is hard to have an index because the base unit is not a simple unit.

For example, the base unit for `newton` is `kilogram-meter-per-square-second`

Reply:
There could be two units with the same base unit `kilogram-meter-per-square-second`. In fact, there are currently three in your output (I guess Newton, lbf, and dyne), meaning the string `kilogram-meter-per-square-second` is included three times. This should be deduplicated.